tournamentcontrol/competition/models.py

```diff
@@ -656,6 +668,50 @@ def get_timeslots(self, date=None):
             rset.rrule(timeslot.rrule())
         return [dt.time() for dt in rset]
 
+    def set_thumbnail_image(self, image_data, mimetype, propagate_to_matches=True):
+        """
+        Set the thumbnail image for this season.
+        
+        Args:
+            image_data (bytes): Binary image data
+            mimetype (str): MIME type of the image
+            propagate_to_matches (bool): Whether to propagate to all matches
+        """
+        self.thumbnail_image = image_data
+        self.thumbnail_image_mimetype = mimetype
+        self.save(update_fields=['thumbnail_image', 'thumbnail_image_mimetype'])
+        
+        if propagate_to_matches:
```

Comment:
Rather than copying the thumbnail to Match, the match should instead take the value of the Season thumbnail if it is not set.